documents_context.md

```diff
@@ -0,0 +1,62 @@
+# Documents Context
+
+## Events
+
+| Type                                            | GARAIO REM         | REM | Description               |
+| ----------------------------------------------- | ------------------ | --- | ------------------------- |
+| [Documents.Document.Add](#documentsdocumentadd) | :white_check_mark: | :x: | Stores a document in GREM |
+
+### Documents.Document.Add
+
+| Field                            | Type     | Content / Remarks                                                              |
+| -------------------------------- | -------- | ------------------------------------------------------------------------------ |
+| `eventType`                      | `string` | `Documents.Document.Add`                                                       |
+| `data`                           | `hash`   |                                                                                |
+| &nbsp;&nbsp;`fetchUrl`           | `string` | an URL where the document can be downloaded from (3) **required**              |
+| &nbsp;&nbsp;`docType`            | `string` | the document type, see (1) **required**                                        |
+| &nbsp;&nbsp;`relatedToType`      | `string` | the type of the business object this document relates to, see (2) **required** |
+| &nbsp;&nbsp;`relatedToReference` | `string` | the reference of the business object this document relates to **required**     |
+| &nbsp;&nbsp;`fileName`           | `string` | the file name **required**                                                     |
+| &nbsp;&nbsp;`description`        | `string` | description of the document                                                    |
+
+Notes
+
+* (1) A generic document type is `Dokument`. To see further type names, perform the _Dokument hochladen_ action in one of the various _Dossier_ UIs in the application. There is a _Dokument-Typ_ dropdown with the type names there. This is a work in progress, [contact us](https://github.com/Garaio-REM/grem-dev-guide) if you have questions regarding document types.
+* (2) Possible values
+  * accounting
+  * property
+  * building
+  * unit
+  * invoicing-invoice (set `docType` to `KreditorRechnung` when delivering an invoice)
+  * letting-tenancy-agreement (consider setting `docType` to `MietvertragDokument` if you're delivering a signed tenancy agreement)
```

Comment:
Now I am fully confused - it seems the docType must be a joint `dokument.type` and with BelegDokument it is `dokument.beleg.class` 

so what do we do with NkaDokument < BelegDokument - type is `NkaDokument` ?